src/components/ConfigEditor.tsx

```diff
@@ -69,6 +79,15 @@ export const BigQueryConfigEditor: React.FC<BigQueryConfigEditorProps> = (props)
             menuShouldPortal={true}
           />
         </Field>
+        <Field label="Custom API endpoint" description={<span>Use a custom API endpoint.</span>}>
+          <Input
+            className="width-30"
+            placeholder="Optional, example https://bigquery.googleapis.com/bigquery/v2/"
+            type={'string'}
+            value={jsonData.customEndpoint || ''}
+            onChange={onCustomEndpointChange}
```

Comment:
Instead of this function you can use `onUpdateDatasourceJsonDataOption` from `@grafana/data` like this:
```suggestion
            onChange={onUpdateDatasourceJsonDataOption(props, 'customEndpoint')}
```